src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -1090,14 +1108,12 @@ private void GossipTo(Address address)
             }
         }
 
-        private Address SelectRandomNode(ImmutableHashSet<Address> addresses)
+        private Address SelectRandomNode(IReadOnlyList<Address> addresses)
         {
             if (addresses.Count > 0)
             {
-                var random = ThreadLocalRandom.Current.Next(addresses.Count - 1);
-                return addresses.Skip(random).First();
+                return addresses[ThreadLocalRandom.Current.Next(addresses.Count)];
```

Comment:
So is this why we're using a list rather than a hash set? To be able to use the indexer?

Reply:
The real input type of `addresses` is `ImmutableSortedSet<>`. The `IReadOnlyList` in this method is to be able to use the index